clients/venice-client/src/main/java/com/linkedin/venice/fastclient/StatsAvroGenericStoreClient.java

```diff
@@ -110,19 +110,13 @@ private <R> CompletableFuture<R> recordRequestMetrics(
       long startTimeInNS,
       FastClientStats clientStats) {
 
+    /**
+     * In case of failures we still attempt to increment the metrics leading up to it.
+     */
     return innerFuture.handle((value, throwable) -> {
       double latency = LatencyUtils.getLatencyInMS(startTimeInNS);
-      if (throwable != null) {
-        clientStats.recordUnhealthyRequest();
-        clientStats.recordUnhealthyLatency(latency);
-        if (throwable instanceof VeniceClientException) {
-          throw (VeniceClientException) throwable;
-        } else {
-          throw new VeniceClientException(throwable);
-        }
-      }
 
```

Comment:
I believe this returns as soon as we get an exception because we want the other metrics to be relevant only for healthy requests. I think that way this is less noisy. @gaojieliu what do you think.

Reply:
Thats true. With these changes, we won't be able to differentiate all these metrics(eg: `longTailRetryTriggered`) for healthy vs unhealthy requests. Also not sure whether its a good idea to add a few new metrics for unhealthy requests (eg: `longTailRetryTriggeredForUnhealthyRequest`) as we seem to have a lot of metrics already. But do we have a lot of metrics in FC as well?

Reply:
I think what you folks said make sense.

Let us try to minimize the total number of metrics if they are not very useful.

Reply:
There are multiple options that can be considered.
1. Leave it as it was before this PR: For unhealthy requests due to exception, only 2 stats (unhealthy counter, and unhealthy latency) will be touched. For healthy requests and unhealthy requests due to high latency, all counters will be included. This is not so clean either as it includes ambiguity (for unhealthy requests with exception vs due to high latency).
2. The change made in this PR: Increment all stats for all requests. The number of stats remain the same as we are not adding any new stats, but these stats get incremented for healthy as well as unhealthy requests which will lead to ambiguity when analyzing the stats as we won't be able to differentiate these stats as part of healthy vs unhealthy ones. (Eg: How many retries happened for healthy vs unhealthy). But it will give us some signals like increase in some counters like retries, etc.
3. Duplicate a subset of these stats as new stats for the unhealthy cases for retry related counters only: `longTailRetryRequestForUnhealthyRequest`, `retryRequestKeyCountForUnhealthyRequest`, `retryRequestSuccessKeyCountForUnhealthyRequest`
4. Duplicate all stats for unhealthy requests. No ambiguity, but more new metrics added.

I am thinking of going with 2 for now. As it will give us some signals on what is happening in the backend rather than not getting any signals. It feels like a good tradeoff between not having any signals or adding more metrics.